tests/ptycho/test_ptycho.py

```diff
@@ -398,6 +482,86 @@ def test_consistent_lstsq_grad_variable_probe(self):
         assert np.all(result.eigen_weights[..., 1:, probes_with_modes:] == 0), (
             "These weights should be unused/untouched "
             "and should have been initialized to zero.")
+        
+    #===========
+    # rPIE tests
+    #===========
+
+    def test_consistent_rpie_poisson_steplength_dominantmode(self):
+        """Check ptycho.solver.rpie for consistency."""
+        params = tike.ptycho.PtychoParameters(
+            psi=self.psi,
+            probe=self.probe,
+            scan=self.scan,
+            algorithm_options=tike.ptycho.RpieOptions(
+                num_batch=5,
+                num_iter=16,
+            ),
+            probe_options=ProbeOptions(),
+            object_options=ObjectOptions(),
+            exitwave_options=ExitWaveOptions( noise_model = "poisson", step_length_usemodes = "dominant_mode" ),
+        )
+        _save_ptycho_result(
+            self.template_consistent_algorithm(
+                data=self.data,
+                params=params,
+            ),
+            f"mpi{self.mpi_size}-rpie_poisson_steplength_dominantmode{self.post_name}",
+        )
+
+    def test_consistent_rpie_poisson_steplength_allmodes(self):
+        """Check ptycho.solver.rpie for consistency."""
+        params = tike.ptycho.PtychoParameters(
+            psi=self.psi,
+            probe=self.probe,
+            scan=self.scan,
+            algorithm_options=tike.ptycho.RpieOptions(
+                num_batch=5,
+                num_iter=16,
+            ),
+            probe_options=ProbeOptions(),
+            object_options=ObjectOptions(),
+            exitwave_options=ExitWaveOptions( noise_model = "poisson", step_length_usemodes = "all_modes" ),
+        )
+        _save_ptycho_result(
+            self.template_consistent_algorithm(
+                data=self.data,
+                params=params,
+            ),
+            f"mpi{self.mpi_size}-rpie_poisson_steplength_allmodes{self.post_name}",
+        )
+
+    def test_consistent_rpie_unmeasured_detector_regions(self):
+        """Check ptycho.solver.rpie for consistency."""
+        params = tike.ptycho.PtychoParameters(
+            psi=self.psi,
+            probe=self.probe,
+            scan=self.scan,
+            algorithm_options=tike.ptycho.RpieOptions(
+                num_batch=5,
+                num_iter=16,
+            ),
+            probe_options=ProbeOptions(),
+            object_options=ObjectOptions(),
+            exitwave_options=ExitWaveOptions( unmeasured_pixels_scaling = 0.90 ),
+        )
+
+        # Define regions where we have missing diffraction measurement data
+        params.exitwave_options.unmeasured_pixels = np.zeros( self.probe.shape[-2:], np.bool_ )
+        params.exitwave_options.unmeasured_pixels[ 100 : 105, : ] = True
+        params.exitwave_options.unmeasured_pixels[ :, 100 : 105 ] = True
+        params.exitwave_options.measured_pixels = np.logical_not( params.exitwave_options.unmeasured_pixels )
```

Comment:
Is unmeasured_pixels float or boolean? Here it is boolean, but in the declaration of ExitwaveOptions, it is a float.

Reply:
I've changed the declaration default datatype to dtype=bool, although floats still work

Reply:
see new commit

Reply:
let me know if this is resolved and can close this part of the conversation.